client/signup/steps/user/index.jsx

```diff
@@ -146,21 +145,12 @@ export class UserStep extends Component {
 		recaptchaClientId: null,
 	};
 
-	componentDidUpdate() {
-		if ( this.props.step?.status === 'completed' ) {
+	componentDidUpdate( prevProps ) {
+		if (
+			prevProps.step?.status !== this.props.step?.status &&
+			this.props.step?.status === 'completed'
+		) {
```

Comment:
This change fixes a bug where the navigation from `/start/user` to the next step (`start/domain`) would be triggered twice, because the component happens to be rerendered twice in the `completed` state. Then you end up with `/start/domain` being twice in the window history, and you'll be confused when clicking the back button. You have to click twice.

This updated code is equivalent to how `useEffect` would work:
```js
useEffect( () => {
  if ( status === 'completed' ) goToNextStep();
}, [ completed ] );
```